reporter/src/main/kotlin/Main.kt

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2017-2018 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.reporter
+
+import com.beust.jcommander.IStringConverter
+import com.beust.jcommander.JCommander
+import com.beust.jcommander.Parameter
+import com.beust.jcommander.ParameterException
+
+import com.here.ort.model.ScanRecord
+import com.here.ort.model.mapper
+import com.here.ort.utils.PARAMETER_ORDER_HELP
+import com.here.ort.utils.PARAMETER_ORDER_LOGGING
+import com.here.ort.utils.PARAMETER_ORDER_MANDATORY
+import com.here.ort.utils.log
+import com.here.ort.utils.printStackTrace
+import com.here.ort.utils.safeMkdirs
+import com.here.ort.utils.showStackTrace
+
+import java.io.File
+
+import kotlin.system.exitProcess
+
+/**
+ * The main entry point of the application.
+ */
+object Main {
+    const val TOOL_NAME = "scanner"
+
+    private enum class ReportFormat {
+        ;
+
+        companion object {
+            /**
+             * The list of all available report formats.
+             */
+            @JvmField
+            val ALL = ReportFormat.values().asList()
+        }
+    }
+
+    private class ReportFormatConverter : IStringConverter<ReportFormat> {
+        override fun convert(name: String): ReportFormat {
+            try {
+                return ReportFormat.valueOf(name.toUpperCase())
+            } catch (e: IllegalArgumentException) {
+                e.showStackTrace()
+
+                throw ParameterException("Report formats must be contained in ${ReportFormat.ALL}.")
+            }
+        }
+    }
+
+    @Parameter(description = "The scan record file to use.",
+            names = ["--scan-record-file", "-s"],
+            order = PARAMETER_ORDER_MANDATORY)
+    private var scanRecordFile: File? = null
```

Comment:
If this is mandatory, why not make it `required = true` and non-nullable which would also safe you the null-check at the end of `main()`?

Reply:
Done.